.github/workflows/documentation.yml

```diff
@@ -8,13 +8,13 @@ jobs:
       - uses: actions/setup-python@v2
       - name: Install dependencies
         run: |
-          pip install sphinx sphinx-autodoc-typehints
+          pip install sphinx sphinx-autodoc-typehints sphinx-rtd-theme tqdm pandas numpy croniter
```

Comment:
This feels awk, if sphinx is a dev dependency can we just use that environment for consistency, what broke with mocking?

Reply:
In the last few versions of docs, backtesting, data, and analysis have all been broken because mocked packages (e.g. pandas, numpy, croniter) can't have type hints with the Union operator, e.g. `def func(arg: pd.Timestamp | None) -> None` unless the package is imported. I opened an issue with sphinx here https://github.com/sphinx-doc/sphinx/issues/11211 to fix that bug, as right now we don't actually have any functions on the [docs page](https://coin-test.github.io/coin-test/reference.html). If the issue ever gets resolved we can remove these imports.

Reply:
I haven't done that because I don't want to compromise our code quality and style for a non-essential part of the CI/CD pipeline, which in this case is the dependency in sphinx.